contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,210 @@
+
+<box type="info">
+    The sandbox above is editable. You can enter your name in the text box and see how the UI is automatically updated. You can also view and edit the code by clicking on the Both button. 
+</box>
+
+The UI of an Angular app is defined in _templates_ which contain a mix of HTML code and HTML-like custom tags.
+  
+The templates used in Angular are _declarative_ because they describe _what_ the view should look like rather than giving instructions on _how_ to construct the view. In the above example, we declare what we want displayed (i.e., name) and Angular renders the view. A noteworthy feature about these templates is their _dynamic behavior_.
+You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is because the UI and the application data are automatically synchronized by Angular as explained [here](https://angular.io/guide/template-syntax#binding-syntax-an-overview).
+
+Along with the the templates, Angular makes use of _Components_. They contain the behavior instructions written in TypeScript language, which will look as follows.  
+
+```typescript
+@Component({
+  selector: 'app-heroes',
+  templateUrl: './heroes.component.html',
+  styleUrls: ['./heroes.component.css']
+})
+export class HeroesComponent {
+  hero: Hero = {
+    name: 'John Doe'
+  };
+  constructor() { }
+}
+```
+
+The `@Component` tag is used to mark a component. It provides the metadata that determines how the component is processed during runtime.
+
+* The use of `selector: app-heroes` is how Angular identifies this component in a template and renders it.  
+* The `hero.name` used in the template, refers to the name property defined in the above component.
+
+Now let us look at what **Dependency Injection**(DI) means. Dependencies are services that a class needs in order to perform it's function. We can visualize this with the help of an example.
+
+<center>
+<img src="dependency.png" width="60%">
+
+_Figure 1. What is a dependency?_ <sup>[source](https://medium.freecodecamp.org/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f)</sup>
+</center>
+
+Dependency injection is a way of providing a class with the required services. The official website claims that the DI framework increases the "efficiency and modularity" of an app. Given below is a an injectable service class.
+ 
+```typescript
+import { Injectable } from '@angular/core';
+import { HEROES } from './mock-heroes';
+@Injectable({
+  providedIn: 'root',
+})
+export class HeroService {
+  getHeroes() { return HEROES; }
+}
+```
+The `HeroesComponent` declared previously, can make use of this `HeroService` by injecting the dependency in it's constructor as a parameter.
```

Comment:
Hi @RonakLakhotia I was asked to look at yours because our chapters have the same structure, and I just happened to notice that this line should be `dependency in its constructor`, not `dependency in it's constructor`. Just a heads up, great write-up otherwise :D

Reply:
@AyushChatto thanks for pointing that out! Will revert this.